File: guides/hotwire-turbo-micronaut-chat/complete/java/src/main/java/example/micronaut/i18n/MessageSourceFactory.java
package example.micronaut.i18n;

import io.micronaut.context.MessageSource;
import io.micronaut.context.annotation.Factory;
import io.micronaut.context.i18n.ResourceBundleMessageSource;
import jakarta.inject.Singleton;

@Factory
class MessageSourceFactory {

    @Singleton
    MessageSource createMessageSource() {
        return new ResourceBundleMessageSource("i18n.messages");
    }
}

File: guides/hotwire-turbo-micronaut-chat/complete/python/src/example/micronaut/i18n/message_source_factory.py
from micronaut.context import MessageSource
from micronaut.context.annotation import Factory
from micronaut.context.i18n import ResourceBundleMessageSource
from jakarta.inject import Singleton


# tag::clazz[]
@Factory  # <1>
class MessageSourceFactory:
    @Singleton  # <2>
    def create_message_source(self) -> MessageSource:
        return ResourceBundleMessageSource("i18n.messages")
# end::clazz[]
